main/lsp/protocol.cc

```diff
@@ -269,6 +269,11 @@ optional<unique_ptr<core::GlobalState>> LSPLoop::runLSP(shared_ptr<LSPInput> inp
 
                 task = move(taskQueue->pendingTasks.front());
                 taskQueue->pendingTasks.pop_front();
+
+                // Test only: Collect counters from other threads into this thread for reporting.
+                if (task->method == LSPMethod::GETCOUNTERS && !taskQueue->counters.hasNullCounters()) {
```

Comment:
This is an unfortunate hack. Normally this thread combines counters every 5 minutes and reports them to statsd, but we need to do this process on-demand in tests. We don't have access to the task queue structure from normal tasks.